templates/base.lds

```diff
@@ -81,6 +81,16 @@ SECTIONS
      * certain core features */
     PROVIDE(__metal_chicken_bit = {{ chicken_bit|default(0) }});
 
+    /* The memory_ecc_scrub bit is used by _entry code to enable/disable
+     * memories scrubbing to zero  */
+    PROVIDE(__metal_eccscrub_bit = {{ eccscrub_bit|default(0) }});
+
+    /* The RAM memories map for ECC scrubbing */
+{% for memory in ram_memories %}
+    PROVIDE( metal_{{ memory.name }}_memory_start = {{ memory.base_hex }} );
+    PROVIDE( metal_{{ memory.name }}_memory_end = {{ memory.base_hex }} + {{ memory.length_hex }} );
```

Comment:
Should these be made private with a `__` prefix?

Also, are we making sure that all possible memories have symbols defined? How will the assembly handle clearing memories if the symbols may or may not be defined?

Reply:
> Should these be made private with a `__` prefix?
> 
> Also, are we making sure that all possible memories have symbols defined? How will the assembly handle clearing memories if the symbols may or may not be defined?

Not sure about the `__`, as some have while other don't. I am incline to leave as is, since user can use the label. 
The linker file will only have the define for the memory the design has. Metal will define all the possible labels but make them as weak

Reply:
Okay, that's pretty clean as far as that can go. That way Metal always has the symbols it needs and the linker script just overrides.

So say we have multiple DTIMs like for a multi-hart design. Will `start` and `end` span all of the DTIMs? They'll have to be contiguous in the memory map.

Reply:
> Okay, that's pretty clean as far as that can go. That way Metal always has the symbols it needs and the linker script just overrides.
> 
> So say we have multiple DTIMs like for a multi-hart design. Will `start` and `end` span all of the DTIMs? They'll have to be contiguous in the memory map.

I believe DTIM/ITM memory are global and continuous. At least for far, base on U74/U54